src/harness.js

```diff
@@ -284,73 +277,108 @@ class Harness {
     options = _.defaults(options, {
-      now: () => new Date(this.getNow()),
-      resolved: false,
       title: undefined,
       user: this.user,
     });
-    
-    const { tasks } = await getNoolsInstances(this.appSettings, options.user, this._state.contacts, this._state.reports, options.now);
+
+    if (options.resolved) {
+      throw Error('getTasks({ resolved: true }) is not supported. See getTaskDocStates() to understand the state of tasks.');
+    }
+
+    if (options.now) {
+      throw Error('getTasks({ now }) is not supported. See setNow() for mocking time.');
+    }
+
+    const tasks = await this.rulesEngineAdapter.fetchTasksFor(options.user, this._state);
+
+    tasks.forEach(task => task.emission.actions.forEach(action => {
+      action.forId = task.emission.forId; // required to hydrate in loadAction
+    }));
+
     return tasks
-      .filter(task => !!options.resolved || !task.resolved)
-      .filter(task => !options.title || task.title === options.title);
+      .filter(task => !options.title || task.emission.title === options.title);
   }
 
-  async getEmittedTargetInstances(options) {
+  /**
+   * Counts the number of task documents grouped by state. [Explanation of task documents and states]{@link https://docs.communityhealthtoolkit.org/core/overview/db-schema/#tasks}
+   *
+   * @param {Object=} options Some options when summarizing the tasks
+   * @param {string} [options.title=undefined] Filter task documents counted to only those with emitted `title` equal to this parameter. Filter is skipped if undefined.
+   * @param {Object} [options.freshTaskDocs=true] When freshTaskDocs is truthy, the task documents will be refreshed prior to counting their states.
+   *
+   * @returns Map with keys equal to task document state and values equal to the number of task documents in that state.
+   * @example
+   * const summary = await countTaskDocsByState({ title: 'my-task-title' });
+   * expect(summary).to.nested.include({
+   *   Complete: 1, // 1 task events were marked as resolved
+   *   Failed: 2,   // 2 task events were not marked as resolved prior to expiring
+   *   Draft: 3,    // 3 task events are in the future
+   * });
+   *
+   */
+  async countTaskDocsByState(options) {
     options = _.defaults(options, {
-      now: () => new Date(this.getNow()),
-      type: undefined,
-      user: this.user,
+      freshTaskDocs: true,
+      title: undefined,
     });
-    
-    const { targets } = await getNoolsInstances(this.appSettings, options.user, this._state.contacts, this._state.reports, options.now);
-    const targetsByUniqId = targets.reduce((prev, curr) => Object.assign(prev, { [curr._id]: curr }), {});
-    return Object.values(targetsByUniqId)
-      .filter(target =>
-        !options.type ||
-        (typeof options.type === 'string' && target.type === options.type) ||
-        (Array.isArray(options.type) && options.type.includes(target.type))
-      );
+
+    if (options.freshTaskDocs) {
```

Comment:
I wasn't sure... 

Part of me thought this info should just be in the response from `getTasks`... but that's a big change since it just returns an array now.  Sounds like your vote is to remove this option?

Reply:
Not sure also. I was looking for clarity.